packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -28,9 +31,32 @@ abstract contract PermissionManager is Initializable {
     /// @notice A special address encoding if a permission is allowed.
     address internal constant ALLOW_FLAG = address(2);
 
+    /// @notice Grant owner flag to check or assign grant ownership right for a permission
+    uint256 internal constant GRANT_OWNER_FLAG = uint256(2);
+
+    /// @notice Revoke owner flag to check or assign revoke ownership right for a permission
+    uint256 internal constant REVOKE_OWNER_FLAG = uint256(4);
+
+    /// @notice Full owner flag to check or assign full ownership rights for a permission
+    uint256 internal constant FULL_OWNER_FLAG = uint256(6);
+
     /// @notice A mapping storing permissions as hashes (i.e., `permissionHash(where, who, permissionId)`) and their status encoded by an address (unset, allowed, or redirecting to a `PermissionCondition`).
     mapping(bytes32 => address) internal permissionsHashed;
 
+    struct Permission {
+        mapping(address => uint256) delegations; // Owners can delegate the permission so delegatees can only grant it one time only.
+        mapping(address => uint256) owners;
+        bool created;
+        uint64 grantCounter;
+        uint64 revokeCounter;
+    }
```

Comment:
add NatSpect

Reply:
done in https://github.com/aragon/osx/pull/604/commits/8c658ea0613795d99b8aff03bec792bceee5efeb